pkg/internal/ebpf/tpinjector/tpinjector.go

```diff
@@ -42,6 +43,21 @@ func (p *Tracer) AllowPID(uint32, uint32, *svc.Attrs) {}
 func (p *Tracer) BlockPID(uint32, uint32) {}
 
 func (p *Tracer) Load() (*ebpf.CollectionSpec, error) {
+
+	if !ebpfcommon.HasHostPidAccess() {
```

Comment:
I'm not sure this is needed. We can run tp injector now in sidecar mode, since we'll be seeing the sidecar traffic only.